Refunct/Refunct.asl

```diff
@@ -9,71 +8,69 @@ state("Refunct-Win32-Shipping")
 	float endPartialSeconds   : 0x1FBF9EC, 0xC0, 0xBC;
 }
 
-startup
-{
-	timer.CurrentTimingMethod = TimingMethod.GameTime;
-	vars.timerModel = new TimerModel { CurrentState = timer };
-	settings.Add("buttons", true, "Split on buttons");
-	for (var index = 1; index <= 37; ++index)
-		settings.Add(index.ToString(), true, "Button " + index.ToString(), "buttons");
+startup {
+	var extraButtons = new Dictionary<int, int> {
+		{7, 2},
+		{10, 2},
+		{18, 2},
+		{26, 3},
+		{28, 2}
+	};
 
-	settings.Add("cubes", false, "Split on cubes");
-	for (var index = 1; index <= 18; ++index)
-		settings.Add("c" + index.ToString(), true, "Cube " + index.ToString(), "cubes");
-}
+	settings.Add("Split on buttons:");
+	settings.Add("Split on cubes:");
+
+	for (int cluster = 1, actualButtonNumber = 1; cluster <= 31; ++cluster, ++actualButtonNumber) {
+		if (extraButtons.ContainsKey(cluster)) {
+			for (int buttonInCluster = 1; buttonInCluster <= extraButtons[cluster]; ++buttonInCluster) {
+				settings.Add("b" + actualButtonNumber, true, "Button " + cluster + "-" + buttonInCluster, "Split on buttons:");
+				settings.SetToolTip("b" + actualButtonNumber, (buttonInCluster == 1 ? "First" : buttonInCluster == 2 ? "Second" : "Third") + " button of " + cluster);
+				if (buttonInCluster < extraButtons[cluster]) ++actualButtonNumber;
+			}
+		} else {
+			settings.Add("b" + actualButtonNumber, true, "Button " + cluster, "Split on buttons:");
+		}
+	}
+
+	for (int cube = 1; cube <= 18; ++cube)
+		settings.Add("c" + cube, false, "Cube " + cube, "Split on cubes:");
 
-init
-{
 	vars.buttons = 0;
-	vars.cubes = 0;
+	vars.timerModel = new TimerModel { CurrentState = timer };
+	
+	timer.CurrentTimingMethod = TimingMethod.GameTime;
 }
 
-update
-{
-	if (current.resets != old.resets)
-	{
-		vars.buttons = current.buttons;
-		vars.cubes = current.cubes;
-		if (current.buttons == 0 && settings.ResetEnabled)
-			vars.timerModel.Reset();
-	}
+update {
+	if (settings.ResetEnabled && old.resets < current.resets && current.buttons == 0)
```

Comment:
Is there a reason to switch to `<` vs `!=`?

Reply:
`resets` can never decrease, so `<` is all that needs to be checked.

Reply:
Fair enough.